spec/solver_spec.rb

```diff
@@ -0,0 +1,8 @@
+require_relative '../solver'
+
+describe Solver do
+  it 'the factorial method returns the factorial of the number provided' do
+    result = Solver.new.factorial(5)
+    expect result.should be 120
+  end
```

Comment:
- [ ] I recommend that you also add tests that handle the following:
  - Negative value which raises an exception as an expectation (When input is a negative value, it should raise an exception).
  - Special case - when input is 0, the result should equal 1